### ISSUE TITLE
Multiple modale on the same page in a programmatic way

### ISSUE THREAD

**Issue by tristanbes** [bootstrap-vue/bootstrap-vue#1427]
Hello,

I'm trying to have one modale per row on a talbe. I found #1424 but it's not working in my case.

On each row on my table, I got a component that renders a modale: 

```vue
<confirmation-modale
    :id="'team_' + row.item.id"
    ...
    :action-route="generateDetachRoute(row.item.id)">
</confirmation-modale>
```

```vue
// inside the confirmation-modale
<button :v-b-modal="id" :class="finalClass">{{ btnLabel }}</button>
    <b-modal
      :id="id"
      ref="confirmationModaleRef"
      :title="trans('confirmation')"
      :ok-title="trans('submit')"
      no-close-on-backdrop
      hide-header-close
      :visible="showModal"
    >
   ...
```

It's either I don't have this code, and all the modale opens when I click on one button, 
and with this code shared, no modale opens. What am I missing please ?

Also, I close this modale programatically, so do I need to customize the `ref` prop to make it unique so It only close 1 modale or not? `this.$refs.<variable>.hide()` ?

**Comment by tmorehouse** [comment]
Remove the leading color from `:v-b-modal="id"` so that is is just `v-b-modal="id"`

Directives automatically assume the value passed to them is a JavaScript expression.

**Comment by tmorehouse** [comment]
Also, since you are creating multiple modals (one per table row), you may want to use the `lazy` prop on the `<b-modal>` so the HTML content is only rendered when the modal is opened.

Another option is to use a single `<b-modal`> below your table, and use a method to populate your modal content and then trigger the modal open programmatically.

**Comment by tristanbes** [comment]
Thank you for clarification.

Is this problem related to vue ? or css ? Once the modale is opened, the fadeout is not applying to the table below (which is another `b-table` component) 

![metadata_title_account_multigaming](https://user-images.githubusercontent.com/346010/33530049-e5c7828a-d879-11e7-9df7-94d1829b7fd7.png)

**Comment by tmorehouse** [comment]
And I just notices that you are also using the `visible` prop when also using the `v-modal` directive.

If all of your modals are tied to the same value vi `visible` you will run into issues.

**Comment by tmorehouse** [comment]
That issues is probably due to to `z-inzex` values, and the fact that your modal is _inside_ the table cells.

**Comment by tristanbes** [comment]
So in that case how to fix that issue ? Because I can't move the rendered DOM for the modale outside the table since it's on the library side, or can I ?

**Comment by tmorehouse** [comment]
You could create a 'reusable" modal, similar to how it is done on the "complete example" in the [`<b-table>` docs](https://bootstrap-vue.js.org/docs/components/table#complete-example).  The example uses a single modal that is re-used by each row.

**Comment by tristanbes** [comment]
Well my modale is not about showing some text, but it contains logic that's why I moved it on a component called `<confirmation-modale>`

Not sure if I was clear at the first place, because I don't understand how your example could sove the problem on the screenshot: I pressed the button `team.detach` just near the modale body, and when the modale appeared, the grey body does not apply to all the page except another `<b-table>` component.

If I understand your example right, it  creates 1 modale and change the content of it dynamically. That's why I don't understand how it could fix this issue. 

I tried playing with z-index, but no value make the grey appear on the full page :(

Thank you for your help
